src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/api/controllers/WidgetController.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.assessrisksandneeds.api.controllers
+
+import com.fasterxml.jackson.annotation.JsonView
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.RoshRiskWidgetDto
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.View
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.services.RiskService
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.services.exceptions.ExternalApiEntityNotFoundException
+
+@RestController
+class WidgetController(
+  private val riskService: RiskService,
+) {
+
+  @RequestMapping(path = ["/risks/crn/{crn}/widget"], method = [RequestMethod.GET])
+  @Operation(description = "Gets rosh summary for crn")
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "403", description = "Unauthorized"),
+      ApiResponse(responseCode = "404", description = "CRN Not Found"),
+      ApiResponse(responseCode = "200", description = "OK")
+    ]
+  )
+  @PreAuthorize("hasAnyRole('ROLE_PROBATION', 'ROLE_CRS_PROVIDER')")
+  fun getRiskSummaryByCrn(
+    @Parameter(description = "CRN", required = true, example = "D1974X")
+    @JsonView(View.SingleRisksView::class)
+    @PathVariable crn: String
+  ): RoshRiskWidgetDto {
+    return try {
+      riskService.getRoshRiskWidgetDataForCrn(crn)
+    } catch (e: Exception) {
```

Comment:
Ideally we shouldn't attempt to catch Exception, but should let the ControllerAdvice implementations handle EntityNotFoundException and the like

Reply:
Removed this and let the controller advice propagate the 404 👍